python/src/store.rs

```diff
@@ -322,6 +322,163 @@ impl PyStore {
+                    .load_graph(
+                        input,
+                        graph_format,
+                        to_graph_name.as_ref().unwrap_or(&GraphName::DefaultGraph),
+                        base_iri,
+                    )
+                    .map_err(map_loader_error)
+            } else if let Some(dataset_format) = DatasetFormat::from_media_type(mime_type) {
+                if to_graph_name.is_some() {
+                    return Err(PyValueError::new_err(
+                        "The target graph name parameter is not available for dataset formats",
+                    ));
+                }
+                self.inner
+                    .load_dataset(input, dataset_format, base_iri)
+                    .map_err(map_loader_error)
+            } else {
+                Err(PyValueError::new_err(format!(
+                    "Not supported MIME type: {}",
+                    mime_type
+                )))
+            }
+        })
+    }
+
+    /// Loads an RDF serialization into the store from a string.
+    ///
+    /// Loads are applied in a transactional manner: either the full operation succeeds or nothing is written to the database.
+    /// The :py:func:`bulk_load` method is also available for much faster loading of big files but without transactional guarantees.
+    ///
+    /// Beware, the full file is loaded into memory.
+    ///
+    /// It currently supports the following formats:
+    ///
+    /// * `N-Triples <https://www.w3.org/TR/n-triples/>`_ (``application/n-triples``)
+    /// * `N-Quads <https://www.w3.org/TR/n-quads/>`_ (``application/n-quads``)
+    /// * `Turtle <https://www.w3.org/TR/turtle/>`_ (``text/turtle``)
+    /// * `TriG <https://www.w3.org/TR/trig/>`_ (``application/trig``)
+    /// * `RDF/XML <https://www.w3.org/TR/rdf-syntax-grammar/>`_ (``application/rdf+xml``)
+    ///
+    /// It supports also some MIME type aliases.
+    /// For example, ``application/turtle`` could also be used for `Turtle <https://www.w3.org/TR/turtle/>`_
+    /// and ``application/xml`` for `RDF/XML <https://www.w3.org/TR/rdf-syntax-grammar/>`_.
+    ///
+    /// :param input: The RDF string data
+    /// :type input: str
+    /// :param mime_type: the MIME type of the RDF serialization.
+    /// :type mime_type: str
+    /// :param base_iri: the base IRI used to resolve the relative IRIs in the file or :py:const:`None` if relative IRI resolution should not be done.
+    /// :type base_iri: str or None, optional
+    /// :param to_graph: if it is a file composed of triples, the graph in which the triples should be stored. By default, the default graph is used.
+    /// :type to_graph: NamedNode or BlankNode or DefaultGraph or None, optional
+    /// :raises ValueError: if the MIME type is not supported or the `to_graph` parameter is given with a quad file.
+    /// :raises SyntaxError: if the provided data is invalid.
+    /// :raises IOError: if an I/O error happens during a quad insertion.
```

Comment:
@Tpt would this still raise an `IOError` anywhere?

Reply:
No, it won't.